### PR TITLE
Feature upgrade pendulum v2

### PR DESCRIPTION
**Pull request by KhaledBousrih**

### CODE REVIEW

**Comment by KhaledBousrih** [comment]
closes #15

File: concrete_datastore/concrete/apps.py [hunk 1]
```diff
@@ -52,12 +57,52 @@ def purge_users_with_wrong_usernames(user_model):
                 archive_legacy(user=user, email_list=email_list)
 
 
+def alter_migration_content(file):
```
Comment (paschembri) [inline concrete_datastore/concrete/apps.py]: Why do we need to alter existing migrations ? I think we can avoid this

File: concrete_datastore/api/v1/datetime.py [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-    return ensure_pendulum(dt).format(api_format, formatter='classic')
+    return ensure_pendulum(dt).format(api_format)
 
 
 def ensure_pendulum(value):
     d = value
     if isinstance(d, str):
         d = pendulum.parse(d)
-    if not isinstance(d, pendulum.pendulum.Pendulum):
+    if not isinstance(d, pendulum.DateTime):
```
Comment (paschembri) [inline concrete_datastore/api/v1/datetime.py]: is there a `pendulum.Date` also ?

File: concrete_datastore/concrete/apps.py (hunk 1 above)
Comment (paschembri) [inline concrete_datastore/concrete/apps.py]: Why do we need to alter existing migrations ? I think we can avoid this

File: concrete_datastore/concrete/apps.py [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
 from collections import defaultdict
 
 from django.utils.translation import ugettext_lazy as _
 from django.apps import AppConfig
+from django.conf import settings
 from django.apps import apps
 
 
 logger_archive_users = logging.getLogger('archive-concrete-users')
+logger_update_pendulum = logging.getLogger('update-pendulum-v2')
```
Comment (paschembri) [inline concrete_datastore/concrete/apps.py]: Adding a logger is not necessary, we can merge loggers : for example a system-logger works fine

File: tests/unittest_settings.py [hunk 4]
```diff
@@ -859,4 +859,8 @@
 
 
 DISABLED_MODELS = ("EntityDividerModel",)
-EMAIL_HOST = ''
+
```
Comment (paschembri) [inline tests/unittest_settings.py]: This file should not be changed by this PR

File: tests/tests_api_v1_1/test_filters.py [hunk 5]
```diff
@@ -1,15 +1,19 @@
 # coding: utf-8
```
Comment (paschembri) [inline tests/tests_api_v1_1/test_filters.py]: Please can you apply only changes due to pendulum upgrade. If this file requires other changes, submit another PR

File: tests/migrations/0001_initial.py [hunk 6]
```diff
@@ -583,7 +583,7 @@ class Migration(migrations.Migration):
                 ('created', models.DateTimeField(auto_now_add=True, verbose_name='Created')),
                 ('expired', models.BooleanField(default=False)),
                 ('expiration_date', models.DateTimeField(default=concrete_datastore.concrete.models.compute_auth_token_expiry)),
-                ('last_action_date', models.DateTimeField(default=pendulum.pendulum.Pendulum.now)),
+                ('last_action_date', models.DateTimeField(default=pendulum.now)),
```
Comment (paschembri) [inline tests/migrations/0001_initial.py]: There may be another solution to modifying migrations

File: tests/tests_api_v1_0/test_filters.py [hunk 7]
```diff
@@ -366,8 +366,7 @@ def setUp(self):
             },
         )
         self.token = resp.data['token']
-        self.date = pendulum.from_format("2017-10-28", '%Y-%m-%d')
-        self.datetime = pendulum.parse("2017-10-28T")
```
Comment (paschembri) [inline tests/tests_api_v1_0/test_filters.py]: `self.datetime` is not used elsewhere ?

File: concrete_datastore/api/v1/datetime.py (hunk 2 above)
Comment (KhaledBousrih) [reply concrete_datastore/api/v1/datetime.py]: yes there is a `pendulum.Date` and in this case it's wiser to use `Date` than `DateTime` because `DateTime` inherits from `Date`

File: concrete_datastore/concrete/apps.py (hunk 3 above)
Comment (KhaledBousrih) [reply concrete_datastore/concrete/apps.py]: in the existing migrations, we may find an import of `pendulum.pendulum` (pendulum < v2.0). With the v2 of pendulum, django commands crash with `ModuleNotFoundError: No module named 'pendulum.pendulum'`

File: tests/unittest_settings.py (hunk 4 above)
Comment (KhaledBousrih) [reply tests/unittest_settings.py]: removed

File: tests/migrations/0001_initial.py (hunk 6 above)
Comment (KhaledBousrih) [reply tests/migrations/0001_initial.py]: removed

File: tests/tests_api_v1_0/test_filters.py (hunk 7 above)
Comment (KhaledBousrih) [reply tests/tests_api_v1_0/test_filters.py]: indeed it is not used

File: concrete_datastore/concrete/apps.py (hunk 1 above)
Comment (KhaledBousrih) [reply concrete_datastore/concrete/apps.py]: in the existing migrations, we may find an import of `pendulum.pendulum` (pendulum < v2.0). With the v2 of pendulum, django commands crash with `ModuleNotFoundError: No module named 'pendulum.pendulum'`

File: tests/tests_api_v1_1/test_filters.py (hunk 5 above)
Comment (KhaledBousrih) [reply tests/tests_api_v1_1/test_filters.py]: removed

File: tests/migrations/0001_initial.py (hunk 6 above)
Comment (lcognat) [reply tests/migrations/0001_initial.py]: What do you mean ?

File: concrete_datastore/api/v1/datetime.py (hunk 2 above)
Comment (paschembri) [reply concrete_datastore/api/v1/datetime.py]: OK

File: concrete_datastore/concrete/apps.py (hunk 1 above)
Comment (paschembri) [reply concrete_datastore/concrete/apps.py]: As this code modifies python source files (migration files), a complete restart is required so we cannot let `alter_migration_content` in the module `concrete_datastore/concrete/apps.py`.
Furthermore, it will require to clear the pyc / __pycache__ and that can be an issue.

Other solutions can be discussed:
- a management command 
- a hook in the migration module to patch migrations at runtime
- no automatic rewrite of migrations but the script can propose a manual action to the developer with the required patch (explicit)

Other (bad) edge cases can arise from modifiying source files from within `concrete`.

File: tests/migrations/0001_initial.py [hunk 8]
```diff
@@ -583,7 +583,7 @@ class Migration(migrations.Migration):
                 ('created', models.DateTimeField(auto_now_add=True, verbose_name='Created')),
                 ('expired', models.BooleanField(default=False)),
                 ('expiration_date', models.DateTimeField(default=concrete_datastore.concrete.models.compute_auth_token_expiry)),
-                ('last_action_date', models.DateTimeField(default=pendulum.now)),
+                ('last_action_date', models.DateTimeField(default=pendulum.pendulum.Pendulum.now)),
```
Comment (paschembri) [inline tests/migrations/0001_initial.py]: Here we need a timezone.now

Comment (KhaledBousrih) [reply tests/migrations/0001_initial.py]: Changed, thanks

File: concrete_datastore/concrete/apps.py (hunk 1 above)
Comment (KhaledBousrih) [reply concrete_datastore/concrete/apps.py]: this will be handeled in another script to fix migrations
